staging/src/k8s.io/apiserver/pkg/server/signal.go

```diff
@@ -67,3 +70,35 @@ func RequestShutdown() bool {
 
 	return false
 }
+
+func dumpStacks(writeToFile bool, path string) {
+	var (
+		buf       []byte
+		stackSize int
+	)
+	bufferLen := 16384
```

Comment:
Why the buff size is `2<<13`? There are thounds of goroutines running in kubelet, what if the buff size is not enough to dump all of them?

> I afraid that use more memory.

Why pprof endpoint will cost more memory? Both approach are all based on `runtime.Stack` to dump goroutine to buff and then convert it to string and then write to a file or tcp fd. 
If you are talking about the overhead of SSL/HTTP,  IMHO, it' insufficient and can be ignored.